src/main/java/org/candlepin/model/ConsumerCloudData.java

```diff
@@ -215,13 +265,80 @@ public ConsumerCloudData setCloudOfferingIds(Collection<String> cloudOfferingIds
         }
 
         if (combinedCloudOfferingIds.length() > CLOUD_OFFERING_ID_MAX_LENGTH) {
-            throw new IllegalArgumentException("cloudOfferingId exceeds the max length");
+            throw new IllegalArgumentException(
+                "Combined cloudOfferingIds exceed the max length of 255 characters");
         }
 
         this.cloudOfferingIds = combinedCloudOfferingIds;
         return this;
     }
 
+    /**
+     * Sets the cloud offering IDs to the specified IDs, replacing any existing IDs.
+     *
+     * <p>This method replaces the current list of cloud offering IDs with the provided IDs.
+     * The IDs are stored internally as a comma-separated string.</p>
+     *
+     * @param cloudOfferingIds
+     *         the cloud offering IDs to be set; must not be {@code null} or contain {@code null} elements;
+     *         if there is empty list it will store {@code null}
+     *
+     * @return a reference to this {@code ConsumerCloudData} instance
+     *
+     * @throws IllegalArgumentException
+     *         if {@code cloudOfferingIds} is {@code null} or contains {@code null} elements,
+     *         or if the combined cloud offering IDs exceed the maximum allowed length of 255 characters
+     */
+    public ConsumerCloudData setCloudOfferingIds(String... cloudOfferingIds) {
+        if (cloudOfferingIds == null) {
+            throw new IllegalArgumentException("cloudOfferingIds is null");
+        }
+
+        this.setCloudOfferingIds(Arrays.asList(cloudOfferingIds));
+        return this;
+    }
+
+    /**
+     * Sets the cloud offering IDs to the specified collection of IDs, replacing any existing IDs.
+     *
+     * <p>This method replaces the current list of cloud offering IDs with the provided collection.
+     * The IDs are stored internally as a comma-separated string.</p>
+     *
+     * @param cloudOfferingIds
+     *         the cloud offering IDs to be set; must not be {@code null},
+     *         or contain {@code null} elements; if there is empty list it will store {@code null}
+     *
+     * @return a reference to this {@code ConsumerCloudData} instance
+     *
+     * @throws IllegalArgumentException
+     *         if {@code cloudOfferingIds} is {@code null}, contains {@code null} elements,
+     *         or if the combined cloud offering IDs exceed the maximum allowed length of 255 characters
+     */
+    public ConsumerCloudData setCloudOfferingIds(Collection<String> cloudOfferingIds) {
+        if (cloudOfferingIds == null) {
+            throw new IllegalArgumentException("cloudOfferingIds is null");
+        }
+        if (cloudOfferingIds.isEmpty()) {
+            this.cloudOfferingIds = null;
+            return this;
+        }
+        for (String cloudOfferingId : cloudOfferingIds) {
```

Comment:
Let's add line breaks between each of these blocks so it's clear each is separate from the others.